src/components/Sing/SequencerGrid.vue

```diff
@@ -112,6 +129,44 @@ const gridWidth = computed(() => {
 const gridHeight = computed(() => {
   return gridCellHeight.value * keyInfos.length;
 });
+// 小節幅
+const measureWidth = computed(() => beatWidth.value * beatsPerMeasure.value);
+// グリッド線の計算
+// オクターブ線や小節線と重なる線は除外
+const gridLines = computed(() => {
+  return keyInfos.reduce<{
+    horizontalLines: number[];
+    octaveLines: number[];
+  }>(
+    (acc, keyInfo, index) => {
+      if (keyInfo.pitch === "F") acc.horizontalLines.push(index);
+      else if (keyInfo.pitch === "C") acc.octaveLines.push(index);
+      return acc;
+    },
+    { horizontalLines: [], octaveLines: [] },
+  );
+});
+const numberOfMeasureLines = computed(() =>
+  Math.ceil(gridWidth.value / measureWidth.value),
+);
+const horizontalLineIndices = computed(() => gridLines.value.horizontalLines);
+const octaveLineIndices = computed(() => gridLines.value.octaveLines);
+const beatLineIndices = computed(() =>
+  Array.from({ length: beatsPerMeasure.value - 1 }, (_, i) => i + 1),
+);
+const snapLinePositions = computed(() => {
+  const snapTicks = gridCellTicks.value;
+  const measureTicks =
+    (tpqn.value * 4 * beatsPerMeasure.value) / timeSignatures.value[0].beatType;
+  const snapPositions = [];
+  let currentTick = 0;
+  while (currentTick < measureTicks) {
+    const x = (currentTick / measureTicks) * measureWidth.value;
+    snapPositions.push(x);
+    currentTick += snapTicks;
+  }
+  return snapPositions;
```

Comment:
（あまり重要度の高くないコメントです！）

whileで書くとワンチャン無限ループするかもなのでちょっと危険かも。
map使うときれいかも？（感覚は人によりそう）
```suggestion
  return Array.from({ length: Math.floor(measureTicks / snapTicks) }, (_, i) => {
    return (i * snapTicks / measureTicks) * measureWidth.value;
  });
```
ちょっと提案コードがあってるかどうか不安ですが。。

Reply:
こちら修正いたしました